src/tagsearch.cc

```diff
@@ -0,0 +1,118 @@
+/********************************************************************
+ * livegrep -- tagsearch.cc
+ * Copyright (c) 2011-2013 Nelson Elhage
+ *
+ * This program is free software. You may use, redistribute, and/or
+ * modify it under the terms listed in the COPYING file.
+ ********************************************************************/
+#include "tagsearch.h"
+#include "content.h"
+#include "lib/debug.h"
+
+#include <utility>
+
+using re2::RE2;
+
+void tag_searcher::load_index(const string& path) {
+    cs_.load_index(path);
+}
+
+void tag_searcher::cache_indexed_files(code_searcher* cs) {
+    file_alloc_ = cs->alloc_;
+    for (auto it = cs->begin_files(); it != cs->end_files(); ++it) {
+        auto file = *it;
+        auto key = repo_path_pair(file->tree->name, file->path);
+        path_to_file_map_.insert(std::make_pair(key, file));
+    }
+}
+
+std::string tag_searcher::create_tag_line_regex(
+    const std::string& name,
+    const std::string& file,
+    const std::string& lno,
+    const std::string& tags) const
+{
+    // full regex match for a tag line created with ctags using format=2.
+    return std::string("^") + name + "\\t" + file + "\\t" + lno + "\\;\\\"\\t" + tags + "$";
+}
+
+std::string tag_searcher::create_partial_regex(RE2 *re) const {
```

Comment:
I'd rather make these helpers normal functions inside an anonymous namespace, as opposed to methods, I think, since they don't actually act on instances.